### PR TITLE
[EPMRPP-38224] Update data in ES in case of removing sub-defects

### PR DESCRIPTION
**Pull request by ihar-kahadouski**

### CODE REVIEW

File: src/main/java/com/epam/ta/reportportal/core/events/handler/DefectTypeDeletedHandler.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+import static com.epam.ta.reportportal.commons.Predicates.equalTo;
+import static com.epam.ta.reportportal.commons.validation.BusinessRule.expect;
+import static com.epam.ta.reportportal.core.analyzer.impl.AnalyzerUtils.getAnalyzerConfig;
+
+/**
+ * @author <a href="mailto:ihar_kahadouski@epam.com">Ihar Kahadouski</a>
+ */
+@Component
+@Transactional(propagation = Propagation.REQUIRES_NEW)
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/events/handler/DefectTypeDeletedHandler.java]: is it needed?

File: src/main/java/com/epam/ta/reportportal/core/events/handler/DefectTypeDeletedHandler.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+		this.projectRepository = projectRepository;
+	}
+
+	@TransactionalEventListener
+	public void handleDfectTypeDeleted(DefectTypeDeletedEvent event) {
+		Project project = projectRepository.findById(event.getProjectId())
+				.orElseThrow(() -> new ReportPortalException(ErrorType.PROJECT_NOT_FOUND, event.getProjectId()));
+
+		if (analyzerServiceClient.hasClients()) {
+			expect(
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/events/handler/DefectTypeDeletedHandler.java]: okey
if there will be running analyzer and the delete defect type event comes
you forbid updating indexing 
results:
analyzer finishes running
defect type is deleted and index is not updated

probably you need to implement some kind of retry logic to wait analyzing finish.
spring retry for example

File: src/main/java/com/epam/ta/reportportal/core/events/handler/DefectTypeDeletedHandler.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+					analyzerStatusCache.getAnalyzeStatus().asMap().containsValue(event.getProjectId()),
+					equalTo(false)
+			).verify(ErrorType.FORBIDDEN_OPERATION, "Index can not be removed until auto-analysis proceeds.");
+
+			List<Long> launches = launchRepository.findLaunchIdsByProjectId(event.getProjectId());
+			logIndexer.deleteIndex(event.getProjectId());
+			logIndexer.indexLogs(event.getProjectId(), launches, getAnalyzerConfig(project));
+		} else {
+			LOGGER.warn("There are no analyzer's.");
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/events/handler/DefectTypeDeletedHandler.java]: you don't need else logic here cause rp without analyzers is ok

File: src/main/java/com/epam/ta/reportportal/core/events/handler/DefectTypeDeletedHandler.java (hunk 1 above)
Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/core/events/handler/DefectTypeDeletedHandler.java]: Yes

Comment (pbortnik) [reply src/main/java/com/epam/ta/reportportal/core/events/handler/DefectTypeDeletedHandler.java]: why?

**Pull request merged**